activerecord/lib/active_record/relation/batches.rb

```diff
@@ -226,13 +231,22 @@ def in_batches(of: 1000, start: nil, finish: nil, load: false, error_on_ignore:
       relation = apply_limits(relation, start, finish, order)
       relation.skip_query_cache! # Retaining the results in the query cache would undermine the point of batching
       batch_relation = relation
+      empty_scope = to_sql == klass.unscoped.all.to_sql
 
       loop do
         if load
           records = batch_relation.records
           ids = records.map(&:id)
           yielded_relation = where(primary_key => ids)
           yielded_relation.load_records(records)
+        elsif (empty_scope && use_ranges != false) || use_ranges
+          ids = batch_relation.pluck(primary_key)
+          finish = ids.last
+          if finish
+            yielded_relation = apply_finish_limit(batch_relation, finish, order)
+            yielded_relation = yielded_relation.except(:limit, :order)
```

Comment:
If I understand it well, this line is now crucial in here and it is missing currently test coverage (I tried to comment it out and tests were still passing).

Would it be possible to add at least comment and test case? Originally I was thinking to move this logic (excluding `LIMIT` and `ORDER` clauses) to actual `ActiveRecord::Batches::BatchEnumerator` since it is useful for methods in there.

Reply:
I believe the `ActiveRecord::Batches::BatchEnumerator` should not modify the relation. 
But for comment and tests - 👍 Please, open a follow up PR if you would like to.

Reply:
resolved by https://github.com/rails/rails/pull/45671 :muscle: